alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt

```diff
@@ -97,7 +97,15 @@ class TransportGetAlertsAction @Inject constructor(
             queryBuilder.filter(QueryBuilders.termQuery("severity", getAlertsRequest.severityLevel))
         }
 
-        if (getAlertsRequest.alertState != "ALL") {
+        if (getAlertsRequest.alertState == "ALL") {
+            // alerting dashboards expects chained alerts and individually executed monitors' alerts to be returned from this api
+            // when invoked with state=ALL. They require that audit alerts are NOT returned in this page
+            // and only be shown in "associated alerts" field under get workflow_alerts API.
+            // But if the API is called with query_params: state=AUDIT,monitor_id=<123>,workflow_id=<abc>, this api
+            // will return audit alerts generated by delegate monitor <123> in workflow <abc>
```

Comment:
If we are still letting this API call audit alerts, it should be included in ALL. It should not be just for the dashboard use case. 

Otherwise, we should not let users see audit alerts through this API and only from the workflow alerts API

Reply:
If not supporting it in `ALL` is temporary until the front end can fix it, then I am fine with going with this logic as long as we have an open github issue about it

Reply:
AUDIT alerts is a state designed to ållow us to create events which we don't want to get users notified about. That's why they rely on chained alerts to be actionable(notifications/acknowledgement etc)

When we return alerts in the api it would clutter the response.

Reply:
Created issue to discuss further and reconcile with front end https://github.com/opensearch-project/alerting/issues/1021

Reply:
We would be introducing breaking changes behavior if we change how it is later on

Reply:
I think we should make a call on if we want audit alerts here or not now.